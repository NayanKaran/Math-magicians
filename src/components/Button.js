import React from 'react';
import PropTypes from 'prop-types';

const Button = (props) => {
  const { buttonProp: { onClick, value } } = props;

  return (
    <button type="button" onClick={onClick} value={value}>
      {value}
    </button>
  );
};

export default Button;

Button.propTypes = {
  buttonProp: PropTypes.shape({
    onClick: PropTypes.func,
    value: PropTypes.string,
  }),
};

Button.defaultProps = {
  buttonProp: null,
};
