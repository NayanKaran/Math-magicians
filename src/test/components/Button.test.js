import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import Button from '../../components/Button';

describe('Testing Button component', () => {
  const buttonProp = { onClick: jest.fn(), value: 'AC' };

  it('Testing with snapshot', () => {
    const tree = renderer.create(<Button buttonProp={buttonProp} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Testing clickEvent', () => {
    render(<Button buttonProp={buttonProp} />);
    fireEvent.click(screen.getByText(buttonProp.value));
    expect(buttonProp.onClick).toHaveBeenCalledTimes(1);
  });
});
