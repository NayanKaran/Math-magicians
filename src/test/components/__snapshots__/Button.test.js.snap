// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Button component Testing with snapshot 1`] = `
<button
  onClick={[MockFunction]}
  type="button"
  value="AC"
>
  AC
</button>
`;
